onnxruntime/contrib_ops/cuda/quantization/qordered_ops/qordered_layer_norm_impl.cu

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/cuda/cu_inc/common.cuh"
+#include "qordered_common.cuh"
+#include "qordered_layer_norm_impl.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace cuda {
+
+constexpr static unsigned QORDER_LAYERNORM_ROWS_PER_BLOCK = 8;
+
+__device__ inline int8_t QuantizeFloatS8(const float val, const float inverse_scale) {
+  float dqval = fmaxf(fminf(127.0f, val * inverse_scale), -128.0f);
+  return static_cast<int8_t>(__float2int_rn(dqval));
+}
+
+// TODO: Const Ref ?
+__device__ inline char4 QuantizeFloat4Char4(const float4 val4, const float rscale) {
+  return char4{QuantizeFloatS8(val4.x, rscale), QuantizeFloatS8(val4.y, rscale),
+               QuantizeFloatS8(val4.z, rscale), QuantizeFloatS8(val4.w, rscale)};
+}
+
+// TODO: Const Ref ?
+__device__ inline int32_t Dp4a_Defined(const char4 input_1, const char4 input_2) {
+  return static_cast<int8_t>(input_1.x) * static_cast<int8_t>(input_2.x) +
```

Comment:
Need to manually define a dot product function as the dp intrinsics are only available for arch >= SM61